python-spec/src/somacore/data.py

```diff
@@ -46,15 +46,26 @@ def create(
         platform_config: Optional[options.PlatformConfig] = None,
         context: Optional[Any] = None,
     ) -> Self:
-        """Creates a new DataFrame."""
+        """Creates a new ``DataFrame`` at the given URI.
```

Comment:
do we want to include info on the requirements & defaulting around soma_joinid?

i.e.,
* soma_joinid must have type pyarrow.int64
* if soma_joinid is omitted from the user-provided schema, it will be added
* it may be an indexed column

Reply:
Added this information.